controllers/overcloudipset_controller.go

```diff
@@ -122,20 +122,35 @@ func (r *OvercloudIPSetReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 			start := net.ParseIP(network.Spec.AllocationStart)
 			end := net.ParseIP(network.Spec.AllocationEnd)
 
-			ip, reservation, err := common.AssignIP(*cidr, start, end, network.Status.Reservations, instance.Name)
+			reservationIP := ""
 
-			// record the reservation on the OvercloudNet
-			network.Status.Reservations = reservation
-			err = r.Client.Status().Update(context.TODO(), network)
-			if err != nil {
-				r.Log.Error(err, "Failed to update OvercloudNet status %v")
-				return ctrl.Result{}, err
+			// Do we already have a reservation for this hostname on the network?
+			for _, reservation := range network.Status.Reservations {
+				if reservation.Hostname == hostname {
+					reservationIP = reservation.IP
```

Comment:
I think with this we want to remove the check in L103-L105 and return the already existing reservation here.
```
			if instance.Status.HostIPs[hostname].IPAddresses[netName] != "" {
				continue
			}
```
I have seen issues with our current version and the above check/continue as ctrlplanCidr might not get set and added to the environment file.

Reply:
Yes, I think that might be right.  If the `hostname` is already present in the OvercloudIPSet status, then we can assume the same reservation is also already stored in the OvercloudNet.  But should we return, or do something else (other than `continue`)?  Do we want the `EnsureConfigMaps` logic to be processed?  A return would skip that logic.

Reply:
atm I think we want it processed with the existing information

Reply:
Agree that you can probably drop the check at line 103 with this new approach. It probably doesn't hurt to recreate the ConfigMaps

Reply:
Sounds good.  I will remove it and run the change through my tests.